### PR TITLE
First importing the sentiment analysis service into the clarity ecosystem

### PR DESCRIPTION
**Pull request by Thomas-Yang**
Current project includes both implementations using NLTK and Senna to perform POS of the natural language sentences. A polarity and score is returned when the natural language text is provided. Also, logging is supported at the server side to keep records of each analyzed text. closes #14

### CODE REVIEW

**Comment by jhauswald** [comment]
- Could you supply a README to describe how to run this, expected output etc? I don't know how to run it!
- typo in thrfit client (you know where!)
- you have a few commented out lines of code, one with your name probably want to get rid of that no?
- received from Notion should be changed to something more general
- code needs linting: 2 spaces, 80 character wrap

Besides the readme and removing commented code looks good. Once it runs I'll merge it in

File: sentiment-analysis/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+$ thrift --gen py sentiment.thrift
+
+### Start the Sentiment Analysis Service
+
+* NLTK version
+** Usage: python SentimentServerNLTK.py [port]
+* Senna version
+** Usage: python SentimentServerSENNA.py PATH-TO-SENNA [port]
+
+NOTE: you need to point the thrift lib to your thrift installation.
```
Comment (jhauswald) [inline sentiment-analysis/README.md]: where do you do this?

File: sentiment-analysis/SentimentClient.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+"""
+This thrift client is created to send natural language texts to the sentiment analysis service, which returns polarity and score for the whole texts. JSON data format is also supported to easily interpret the results.
+
+@author: Hailong Yang (hailong@umich.edu)
+
+"""
+
+import sys, glob, os
+sys.path.append('gen-py')
+sys.path.insert(0, glob.glob('lib/py/lib.*')[0])
```
Comment (jhauswald) [inline sentiment-analysis/SentimentClient.py]: this line breaks for me, is this where you point the thrift lib?

Comment (Thomas-Yang) [inline sentiment-analysis/SentimentClient.py]: > On Jul 1, 2015, at 1:56 AM, Johann Hauswald <notifications@github.com> wrote:
> 
> > +sys.path.append('gen-py’)
Here you point to the glue code directory generated by thrift.
> +sys.path.insert(0, glob.glob('lib/py/lib.*')[0])
Here you point to the thrift lib for python.
I will add the description into README.

**Comment by jhauswald** [comment]
- I had trouble with those lines of adding thrift to path, why not just do `sudo pip install thrift`? worked for me and I just got rid of that insert line
- I had to run some nltk.download(), should that be in the readme?
- other than that it worked, it needs a sample input, i copy pasted something random from the internet and got a positive review, pretty cool.

After sample input and a `requirements.txt` [file](https://pip.pypa.io/en/latest/user_guide.html#requirements-files) for thrift, lets merge it.

**Comment by Thomas-Yang** [comment]
Cool! I usually build thrift from source, but it is much easier to install from pip. I will change that in the readme. Yes, nltk.download() is necessary to get all the vocabularies as well as dictionaries for NTLK. That shall be added too. Is a separate requirements file necessary? It is not too much to be included into the readme. What do you think? I can prepare a sample input as well.

**Comment by jhauswald** [comment]
I agree maybe a simple line in the readme is good, requirements is when there many. Yes sample input would be great

**Comment by Thomas-Yang** [comment]
I have finished the update. Try you can get it work this time.

**Comment by jhauswald** [comment]
Yup works great with your sample input, merged

**Comment by Thomas-Yang** [comment]
Yeah~

**Pull request merged**